app/src/main/java/org/simple/clinic/facility/change/FacilityChangeScreenController.kt

```diff
@@ -72,9 +73,17 @@ class FacilityChangeScreenController @Inject constructor(
                     .associateUserWithFacility(it, facility)
                     .andThen(facilityRepository.setCurrentFacility(it, facility))
               }
-              .andThen(reportsRepository.deleteReportsFile().toCompletable())
-              .andThen(reportsSync.sync().onErrorComplete())
+              .doOnComplete { clearAndSyncReports() }
               .andThen(Single.just(Ui::goBack))
         }
   }
+
+  private fun clearAndSyncReports() {
+    reportsRepository
+        .deleteReportsFile()
+        .toCompletable()
+        .andThen(reportsSync.sync().onErrorComplete())
+        .subscribeOn(Schedulers.io())
```

Comment:
This was discussed offline, but adding here for records: 

Reports Sync is a background operation which we want to complete regardless of whether the screen is destroyed or not. This is why we run it as a side-effect instead of as part of the event loop. In addition, we run it on a new thread because this could potentially be a long running operation since it involves network calls, and we do not want to wait until it completes to close the screen.